test/e2e/service.go

```diff
@@ -504,15 +504,22 @@ var _ = Describe("Services", func() {
 		By("hitting the pod through the service's LoadBalancer")
 		testLoadBalancerReachable(ingress1, 80)
 
-		By("changing service " + serviceName + " update NodePort")
-		nodePort2 := nodePort1 - 1
-		if !ServiceNodePortRange.Contains(nodePort2) {
-			//Check for (unlikely) assignment at bottom of range
-			nodePort2 = nodePort1 + 1
+		By("changing service " + serviceName + ": update NodePort")
+		nodePort2 := 0
+		for i := 1; i < ServiceNodePortRange.Size; i++ {
+			offs1 := nodePort1 - ServiceNodePortRange.Base
```

Comment:
If we decide that it's too complicated, we can fix that in a follow-up PR, I guess.